airflow/ti_deps/deps/prev_dagrun_dep.py

```diff
@@ -44,25 +44,16 @@ def _get_dep_statuses(self, ti, session, dep_context):
             return
 
         # Don't depend on the previous task instance if we are the first task
-        dag = ti.task.dag
-        if dag.catchup:
-            if dag.previous_schedule(ti.execution_date) is None:
-                yield self._passing_status(reason="This task does not have a schedule or is @once")
-                return
-            if dag.previous_schedule(ti.execution_date) < ti.task.start_date:
```

Comment:
This conditional was checking something like this case:

```python
with DAG(start_date='2020-01-01'):
    task1 = Operator()
    task2 = Operator(start_date='2021-07-01', depends_on_past=True)
```

I guess this isn't covered by the tests either :(

Reply:
Since we don’t really care about the previous schedule, we can still use `DagRun.get_previous_scheduled_dagrun()` here, and use its execute date (or something) instead of `DAG.previous_schedule()`.